### ISSUE TITLE
Eagerloading .with(['parent']) only loads first entry parent

### ISSUE THREAD

**Issue by leevigraham** [craftcms/cms#6146]
### Description

Eagerloading with parent only returns parent for first structure element in loop

Template code:

```twig
<ul class="list-decimal text-xs">
{% for entry in craft.entries().section('tikiMugs').order('structureId').with(['parent']).search('munktiki alibi logo').all() %}
    <li>{{ entry.title }} - {{ entry.parent ? entry.parent.title : 'none' }}</li>
{% endfor %}
</ul>

<ul class="list-decimal text-xs mt-4">
{% for entry in craft.entries().section('tikiMugs').order('structureId').search('munktiki alibi logo').all() %}
    <li>{{ entry.title }} - {{ entry.parent ? entry.parent.title : 'none' }}</li>
{% endfor %}
</ul>
```

Output:

![image](https://user-images.githubusercontent.com/25124/83014326-d8ee7700-a061-11ea-9f47-399b778fc8cc.png)

### Steps to reproduce

1. Create a structure
2. Eager load parents in template

### Additional info

- Craft version: 3.5.beta.2
- PHP version: 7.4
- Database driver & version: mysql
- Plugins & versions:

Could be related to: https://github.com/craftcms/cms/issues/1375

**Comment by brandonkelly** [comment]
Whoops, thanks for reporting. Fixed for the next 3.5 beta release.

**Comment by leevigraham** [comment]
@brandonkelly Just pulled in the latest 3.5 beta to my fork and this issue still persists.

Checked I'm running the latest code:

![image](https://user-images.githubusercontent.com/25124/83208324-b743db80-a198-11ea-9888-a5cd093e599a.png)

Cleared the cache and the output is still the same:

![image](https://user-images.githubusercontent.com/25124/83208352-cf1b5f80-a198-11ea-8f21-efe487b90ddb.png)
